### PR TITLE
Update the minimum external LLVM to 15

### PR DESCRIPTION
**Pull request by cuviper**
With this change, we'll have stable support for LLVM 15 through 17 (pending release).
For reference, the previous increase to LLVM 14 was #107573.

### CODE REVIEW

**Comment by Kobzol** [comment]
https://github.com/rust-lang/rust/blob/6c7f30114515111c53224e2de1434a324f73c0e8/src/ci/scripts/install-clang.sh#L13 this will also probably need changing.

File: tests/ui/dyn-star/llvm-old-style-ptrs.rs [hunk 1]
```diff
@@ -1,8 +1,6 @@
 // run-pass
 // compile-flags: -Copt-level=0 -Cllvm-args=-opaque-pointers=0
```
Comment (nikic) [inline tests/ui/dyn-star/llvm-old-style-ptrs.rs:2]: This test can be deleted entirely. I guess that can be done as part of rustc_codegen_llvm cleanups to drop typed pointer handling (which will probably break this tests).

**Comment by nikic** [comment]
@bors r+

> this will also probably need changing.

I believe this is referring to a host clang version. Not related to LLVM used by rustc.

**Comment by Kobzol** [comment]
> I believe this is referring to a host clang version. Not related to LLVM used by rustc.

Ah, you are right, my bad.

File: tests/ui/dyn-star/llvm-old-style-ptrs.rs (hunk 1 above)
Comment (cuviper) [reply tests/ui/dyn-star/llvm-old-style-ptrs.rs:2]: There's a related line here:
https://github.com/rust-lang/rust/blob/9339f446a5302cd5041d3f3b5e59761f36699167/compiler/rustc_codegen_ssa/src/base.rs#L274

And yeah, I think dropping typed pointers entirely can be a later followup.

**Pull request merged**